control-plane/api-gateway/gatekeeper/role.go

```diff
@@ -83,3 +82,62 @@ func (g *Gatekeeper) role(gateway gwv1beta1.Gateway, gcc v1alpha1.GatewayClassCo
 
 	return role
 }
+
+func (g *Gatekeeper) upsertOpenshiftRole(ctx context.Context, gateway gwv1beta1.Gateway, config common.HelmConfig) error {
```

Comment:
We basically create a separate role and rolebinding for openshift. I think I can refactor these to just be in the "normal" role and role-binding, but didn't know if we wanted them to be separate?

Reply:
No need to have a separate role for openshift, we can just create and modify a single role

Reply:
removed